src/python/pants/backend/experimental/python/lockfile.py

```diff
@@ -259,5 +291,54 @@ async def generate_all_tool_lockfiles(
     return ToolLockGoal(exit_code=0)
 
 
+BEGIN_LOCKFILE_HEADER = b"# --- BEGIN LOCKFILE METADATA: DO NOT EDIT OR REMOVE ---"
+END_LOCKFILE_HEADER = b"# --- END LOCKFILE METADATA ---"
+
+
+# Lockfile metadata for headers
+def validated_lockfile_content(req: PythonLockfileRequest, content: bytes) -> bytes:
```

Comment:
Naming suggestion: normally I think of `validate` as something that will return `None` and only error if invalid. Maybe something like `add_lockfile_header` would work?

Reply:
I went with `validated` as opposed to `validate`; I feel like `add_lockfile_header` reads like something that amends the content

Reply:
`prepend_lockfile_header`? It is amending the content, only in a functional programming style of returning a new value rather than mutating the old.

Reply:
I went for `lockfile_content_with_header`.